### ISSUE TITLE
HTTP2: Should we expose settings on SocketsHttpHandler for inbound flow control?

### ISSUE THREAD

**Issue by geoffkizer** [dotnet/corefx#31299]
HTTP2 allows us to define our window size using the SETTINGS_INITIAL_WINDOW_SIZE setting. The default is 64K.

Currently we simply always use the default. Should we expose a setting to control this?

Note that this setting would basically control the receive buffer size for a response body. 

Flow control is managed both per-stream and per-connection, but the interesting case here is per-stream.

**Comment by karelz** [comment]
Related to #35388 
It is an escape route if we don't get defaults right + involves new API. Not critical for 3.0 at this moment.

**Comment by Tratcher** [comment]
Yes these should be configurable in case people run into throughput or memory issues.

**Comment by scalablecory** [comment]
I think we can let the user configure a maximum receive window size, and then implement some logic to auto-discover it.

**Comment by karelz** [comment]
Triage: We should explore what winsock does.

**Comment by scalablecory** [comment]
Related: https://github.com/aspnet/AspNetCore/issues/4734
